enhancements/network/egress-ip-per-destinations.md

```diff
@@ -0,0 +1,327 @@
+	// matched by the NamespaceSelector. In case it is set: is intersected with
+	// the NamespaceSelector, thus applying the egress IP to the pods
+	// (in the namespace(s) already matched by the NamespaceSelector) which
+	// match this pod selector.
+	// +optional
+	PodSelector metav1.LabelSelector `json:"podSelector,omitempty"`
+}
+```
+
+New CRD:
+
+```golang
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:nostatus
+// +kubebuilder:resource:shortName=eipd,scope=Cluster
+// +kubebuilder:printcolumn:name="DestinationNetworks",type=string,JSONPath=".spec.destinationNetworks"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// EgressIPTraffic defines a set of networks outside the cluster network.
+type EgressIPTraffic struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec EgressIPTrafficSpec `json:"spec"`
+}
+
+// EgressIPTrafficSpec defines the desired state description of EgressIPTraffic.
+// +kubebuilder:validation:MaxProperties=1
+// +kubebuilder:validation:MinProperties=1
+type EgressIPTrafficSpec struct {
+	// DestinationNetworks is the list of Network CIDRs that defines external destinations
+	// for IPv4 or IPv6. Mixing IP families are not allowed.
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	DestinationNetworks []CIDR `json:"destinationNetworks,omitempty" validate:"omitempty,dive,cidr"`
+}
+
+// CIDR is a network CIDR. IPv4 or IPv6.
+// +kubebuilder:validation:XValidation:rule="isCIDR(self)",message="Invalid CIDR"
+type CIDR string
+
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// EgressIPTrafficList contains a list of EgressIPTraffic
+type EgressIPTrafficList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []EgressIPTraffic `json:"items"`
+}
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+Nothing new that's specific for HCP.
+
+#### Standalone Clusters
+Nothing specific to standalone.
+
+#### Single-node Deployments or MicroShift
+Nothing specific for these deployments.
+
+### Implementation Details/Notes/Constraints
+
+#### Cluster manager
+No changes to `EgressIP` node assignment or health checks therefore no changes.
+
+#### OVNKube-controller (OVNKube-node)
+If a user defines an `EgressIP` CR and doesn't populate `TrafficSelector` then `EgressIP` behaviour is the same as today (without this proposed feature).
+All traffic egress-ing the cluster will be SNAT'd.
+
+If a user defines an `EgressIP` CR and populates the `TrafficSelector` field with a selector, and it matches or does not match an `EgressIPTraffic`
+CR but regardless, there are no defined networks among the selected `EgressIPTraffic` CRs, then no cluster external traffic is SNAT'd. This
```

Comment:
Is this also the same case as the question @trozet asked above about a packet which does not fall within any CIDR in any selected EgressIPTraffic CRs? In this case there is no SNAT due to EgressIP but the packet may (would?) be SNAT to the node's interface IP as it egresses?

Reply:
Yes, its sent out the primary interface and SNAT'd to that IP.